### PR TITLE
GitHub Actions CI jobs

### PR DESCRIPTION
**Pull request by pdyraga**
Depends on #1 

The following jobs are defined:
- contracts build and test,
- contracts linting for `.sol` code and `.js` tests,
- slither static analyzer for contracts.

This solution does not cache `node_modules` and it needs to be addressed separately. 

Apologies for force-pushes, this is still a working draft and I need to test things.

### CODE REVIEW

File: solidity/.github/workflows/contracts.yml [hunk 1]
```diff
@@ -0,0 +1,122 @@
+name: Solidity
```
Comment (nkuba) [inline solidity/.github/workflows/contracts.yml:1]: We need to move the configuration file to `.github/workflows/contracts.yml`.

Comment (pdyraga) [reply solidity/.github/workflows/contracts.yml:1]: 🤦

File: .github/workflows/contracts.yml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12.x"
```
Comment (nkuba) [inline .github/workflows/contracts.yml:43]: If we're starting with a new repo, could we give version `14` a try?

File: solidity/.github/workflows/contracts.yml (hunk 1 above)
Comment (nkuba) [reply solidity/.github/workflows/contracts.yml:1]: ... and remove this file :)

File: .github/workflows/contracts.yml (hunk 2 above)
Comment (pdyraga) [reply .github/workflows/contracts.yml]: Sure, why not.

File: .github/workflows/contracts.yml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        with:
+          node-version: "14.x"
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8.5
+
+      - name: Install Solidity
+        env:
+          SOLC_VERSION: 0.7.6 # according to solidity.version in hardhat.config.js
```
Comment (nkuba) [inline .github/workflows/contracts.yml:97]: We should extract the version from `hardhat.config.js` and use it here as a variable. But it's for the future...

**Review by nkuba — APPROVED** [review]

**Pull request merged**